gef-ui/server/controllers/unissued-facilities/index.test.js

```diff
@@ -0,0 +1,383 @@
+
+    api.getApplication.mockResolvedValue({});
+    api.getFacility.mockResolvedValue(mockFacilityResponse);
+    api.updateFacility.mockResolvedValue({});
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const now = new Date();
+  const tomorrow = add(now, { days: 1 });
+  const threeMonthsAndOneDayFromNow = add(now, { months: 3, days: 1 });
+  const oneYearFromNow = add(now, { years: 1, months: 3, days: 1 });
+
+  it('posts and returns correct message and url', async () => {
+    mockRequest.body.facilityType = 'CASH';
+    mockRequest.body.facilityName = 'Foundry4';
+    mockRequest.query.saveAndReturn = 'true';
+    mockRequest.body['issue-date-day'] = format(tomorrow, 'd');
+    mockRequest.body['issue-date-month'] = format(tomorrow, 'M');
+    mockRequest.body['issue-date-year'] = format(tomorrow, 'yyyy');
+
+    mockRequest.body['cover-start-date-day'] = format(tomorrow, 'd');
+    mockRequest.body['cover-start-date-month'] = format(tomorrow, 'M');
+    mockRequest.body['cover-start-date-year'] = format(tomorrow, 'yyyy');
+
+    mockRequest.body['cover-end-date-day'] = format(tomorrow, 'd');
+    mockRequest.body['cover-end-date-month'] = format(tomorrow, 'M');
+    mockRequest.body['cover-end-date-year'] = format(tomorrow, 'yyyy');
+
+    await postChangeUnissuedFacility(mockRequest, mockResponse);
+
+    expect(api.updateFacility).toHaveBeenCalledWith('xyz', {
+      coverEndDate: format(tomorrow, 'MMMM d, yyyy'),
+      coverStartDate: format(tomorrow, 'MMMM d, yyyy'),
+      issueDate: format(tomorrow, 'MMMM d, yyyy'),
+      shouldCoverStartOnSubmission: null,
+      monthsOfCover: null,
+      name: 'Foundry4',
+      hasBeenIssued: true,
+      changedToIssued: true,
+      coverDateConfirmed: true,
+    },
+    { message: 'Foundry4 is updated' },
+    '/gef/application-details/123/unissued-facilities');
+  });
+
+  it('should not update facility if incorrect start and end dates', async () => {
+    mockRequest.body.facilityType = 'CASH';
+    mockRequest.body.facilityName = 'Foundry4';
+    mockRequest.query.saveAndReturn = 'true';
+
+    mockRequest.body['issue-date-day'] = format(tomorrow, 'd');
+    mockRequest.body['issue-date-month'] = format(tomorrow, 'M');
+    mockRequest.body['issue-date-year'] = format(tomorrow, 'yyyy');
+
+    mockRequest.body['cover-start-date-day'] = format(threeMonthsAndOneDayFromNow, 'd');
+    mockRequest.body['cover-start-date-month'] = format(threeMonthsAndOneDayFromNow, 'M');
+    mockRequest.body['cover-start-date-year'] = format(threeMonthsAndOneDayFromNow, 'yyyy');
+
+    mockRequest.body['cover-end-date-day'] = format(oneYearFromNow, 'd');
+    mockRequest.body['cover-end-date-month'] = format(oneYearFromNow, 'M');
+    mockRequest.body['cover-end-date-year'] = format(oneYearFromNow, 'yyyy');
+
+    await postChangeUnissuedFacility(mockRequest, mockResponse);
+    // should not go ahead with call as errors
+    expect(api.updateFacility).not.toHaveBeenCalledWith('xyz', {
```

Comment:
instead of hardcoding xyz can we reference the facilityId via the mock so it's clearer?

Reply:
done